de.dlr.sc.virsat.team/src/de/dlr/sc/virsat/team/git/VirSatGitVersionControlBackend.java

```diff
@@ -93,41 +121,87 @@ protected void doCommit(Repository gitRepository, String message, SubMonitor mon
 
 		commitAndPullMonitor.split(1).subTask("Check if remotes exist");
+		
 		// Get the remotes for the repository
 		String remoteUrl = gitRepository.getConfig().getString("remote", "origin", "url");
 		Collection<Ref> refs = Git.lsRemoteRepository()
 			.setRemote(remoteUrl)
 			.call();
 		
 		// Only perform a pull of the remote exists
-		ProgressMonitor gitMonitor = new EclipseGitProgressTransformer(commitAndPullMonitor.split(1));
-		
 		if (!refs.isEmpty()) {
-			// Pull from origin
-			Git.wrap(gitRepository).pull()
+			
+			// Pull from origin and apply the Recursive Merge Strategy. It is the git standard merge strategy.
+			// In case there has been another commit on the remote in between, the pull will try to resolve it
+			// and store it as a new commit locally.
+			ProgressMonitor gitPullMonitor = new EclipseGitProgressTransformer(commitAndPullMonitor.setWorkRemaining(1).split(1));
+			lastPullResult = Git.wrap(gitRepository).pull()
 				.setCredentialsProvider(credentialsProvider)
-				.setProgressMonitor(gitMonitor)
+				.setProgressMonitor(gitPullMonitor)
 				.setTimeout(GIT_REMOTE_TIMEOUT)
-				.setStrategy(MergeStrategy.THEIRS)
+				.setStrategy(MergeStrategy.RECURSIVE)
 				.call();
-		}
-		
+			
+			checkAndResolveConflicts(gitRepository, commitAndPullMonitor);
+		} 
+		 
 		Git.wrap(gitRepository).close();
 		
 		project.refreshLocal(Resource.DEPTH_INFINITE, commitAndPullMonitor.split(1));
 	}
+	
+	/**
+	 * In case not all files could be merged they need to be fixed.
+	 * Conflicting files are fixed by replacing them with the version from the remote git repository.
+	 * @param gitRepository the git repository
+	 * @param monitor the progress monitor
+	 * @throws Exception
+	 */
+	private void checkAndResolveConflicts(Repository gitRepository, SubMonitor progressMonitor) throws Exception {
+		// Therefore lets see if files are in a conflicting state within the repository
+		Set<String> conflictingFiles = Git.wrap(gitRepository).status().call().getConflicting();
+		
+		// If there are any continue from here with resolving the conflicts.
+		if (!conflictingFiles.isEmpty()) {
+			progressMonitor.setWorkRemaining(2);
+			ProgressMonitor gitCheckoutMonitor = new EclipseGitProgressTransformer(progressMonitor.split(1));
+			
+			// Prepare a checkout command to get THEIRS on all conflicting files.
+			// This basically means that in case of concept the files in the repository
+			// are regarded as being correct. This happens on the cost of loosing local changes.
+			CheckoutCommand checkoutCommand = Git.wrap(gitRepository).checkout();
+			checkoutCommand.setProgressMonitor(gitCheckoutMonitor);
+			
+			for (String conflictingFile : conflictingFiles) {
+				checkoutCommand.addPath(conflictingFile);
+			}
+			checkoutCommand.setStage(Stage.THEIRS);
+			checkoutCommand.call();
+			
+			// Finally close the still open merge commit
+			doCommit(gitRepository, BACKEND_REPOSITORY_MERGE_COMMIT_MESSAGE, progressMonitor.split(1));

```

Comment:
Do we maybe want to notify the user about the replacement? Like with a simple dialog? I'm afraid people will not realize that the change was reverted...

Reply:
@franzTobiasDLR Its not much but in the EGit dialog you at least see the status notification that there were conflicts.